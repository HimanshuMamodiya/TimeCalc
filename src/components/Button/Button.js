import React from 'react'
// import './Button.css'
// import 'material-design-lite/src/button/button'
import PropTypes from 'prop-types'

import Button from '@material-ui/core/Button'

const GenericButton = ({ invoke, context, classes, icon, type }) => (
  <Button
    disabled={!invoke}
    variant='raised'
    color='primary'
    className={classes}
    type={type || 'button'}
    onClick={() => type ? true : invoke(context)}
  >
    <i className='material-icons'>{icon}</i>
  </Button>
)

GenericButton.propTypes = {
  invoke: PropTypes.func,
  context: PropTypes.any,
  classes: PropTypes.string,
  icon: PropTypes.string,
  type: PropTypes.string
}

export default GenericButton
