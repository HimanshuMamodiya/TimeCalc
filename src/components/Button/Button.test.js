import React from 'react'
import Button from './Button'
import { shallow } from 'enzyme'
import 'jest-enzyme'

describe('Component - Button', () => {
  it('adds the classes from props.classes to the button', () => {
    const component = shallow(<Button classes='mdl-button--raised' />)

    expect(component.find('button')).toHaveClassName('mdl-button--raised')
  })

  it('adds the mdl-icon from props.icon', () => {
    const component = shallow(<Button icon='add' />)

    expect(component.find('i')).toHaveHTML("<i class='material-icons'>add</i>")
  })

  it('renders a disabled button if there is no callback defined via props.invoke', () => {
    const component = shallow(<Button />)

    expect(component.find('button')).toBeDisabled()
  })

  it('binds the onClick handler', () => {
    const cb = jest.fn()
    const component = shallow(<Button invoke={cb} context={1337} />)

    component.find('button').simulate('click')
    expect(cb).toBeCalledWith(1337)
  })
})
