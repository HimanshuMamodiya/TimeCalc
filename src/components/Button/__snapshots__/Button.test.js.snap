// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Button adds the classes from props.classes to the button 1`] = `"<button tabindex=\\"-1\\" class=\\"MuiButtonBase-root-17 MuiButtonBase-disabled-18 MuiButton-root-1 MuiButton-raised-7 MuiButton-raisedPrimary-8 MuiButton-disabled-11 foo\\" type=\\"button\\" disabled=\\"\\"><span class=\\"MuiButton-label-2\\"><i class=\\"material-icons\\"></i></span></button>"`;

exports[`Component - Button adds the mdl-icon from props.icon 1`] = `"<button tabindex=\\"-1\\" class=\\"MuiButtonBase-root-17 MuiButtonBase-disabled-18 MuiButton-root-1 MuiButton-raised-7 MuiButton-raisedPrimary-8 MuiButton-disabled-11\\" type=\\"button\\" disabled=\\"\\"><span class=\\"MuiButton-label-2\\"><i class=\\"material-icons\\">add</i></span></button>"`;

exports[`Component - Button renders a disabled button if there is no callback defined via props.invoke 1`] = `"<button tabindex=\\"-1\\" class=\\"MuiButtonBase-root-17 MuiButtonBase-disabled-18 MuiButton-root-1 MuiButton-raised-7 MuiButton-raisedPrimary-8 MuiButton-disabled-11\\" type=\\"button\\" disabled=\\"\\"><span class=\\"MuiButton-label-2\\"><i class=\\"material-icons\\"></i></span></button>"`;

exports[`Component - Button sets the type from props.type 1`] = `"<button tabindex=\\"-1\\" class=\\"MuiButtonBase-root-17 MuiButtonBase-disabled-18 MuiButton-root-1 MuiButton-raised-7 MuiButton-raisedPrimary-8 MuiButton-disabled-11\\" type=\\"reset\\" disabled=\\"\\"><span class=\\"MuiButton-label-2\\"><i class=\\"material-icons\\"></i></span></button>"`;
